jaq-fmts/src/read/formats.rs

```diff
@@ -48,6 +49,17 @@ pub fn from_bufread<'a>(fmt: Format, read: impl BufRead + 'a, s: &'a str, slurp:
     }
 }
 
+fn parse_many_null_terminated(bytes: &Bytes) -> impl Iterator<Item = io::Result<Val>> + '_ {
+    bytes
+        .split_inclusive(|&b| b == b'\0')
```

Comment:
Can you use [`split_str`](https://docs.rs/bstr/latest/bstr/trait.ByteSlice.html#method.split_str) here to simplify this?

Reply:
`split` and `split_str` splits on separators, not terminators, meaning that the last NUL character would introduce a new empty value into the stream.

`split_inclusive` on the other hand will not yield that empty last string after encountering the last terminator.